src/fs/reader.ts

```diff
@@ -0,0 +1,27 @@
+import { IParserResult } from '@stoplight/types';
+import { parseWithPointers } from '@stoplight/yaml';
+import { existsSync, readFileSync } from 'fs';
+// @ts-ignore
+import * as fetch from 'node-fetch';
+
+async function doRead(name: string, encoding: string) {
+  if (name.startsWith('http')) {
+    const result = await fetch(name);
+    return parseWithPointers(await result.text());
+  } else if (existsSync(name)) {
+    try {
+      return parseWithPointers(readFileSync(name, encoding));
+    } catch (ex) {
+      throw new Error(`Could not read ${name}: ${ex.message}`);
+    }
+  }
+  throw new Error(`${name} does not exist`);
```

Comment:
How about check for valid yaml file and one more exception thrown? Parser should return diagnostics array with exceptions.

Reply:
I don't think it's needed given the fact we expose parsing errors in both CLI mode and programmatic.
IHMO we shouldn't bail out if there is a syntax error, but we def need to display error messages in some way - as mentioned, it's already implemented, so I think we are good.

Reply:
Yup, I agree with JR :)